mne/io/edf/edf.py

```diff
@@ -1440,9 +1440,7 @@ def _read_gdf_header(fname, exclude, include=None):
                     n_events = np.fromfile(fid, UINT32, 1)[0]
                 else:
                     ne = np.fromfile(fid, UINT8, 3)
-                    n_events = ne[0]
-                    for i in range(1, len(ne)):
-                        n_events = n_events + ne[i] * 2 ** (i * 8)
+                    n_events = ne[0] + (ne[1] << 8) + (ne[2] << 16)
```

Comment:
I think this is a fix on NumPy < 2 but on 2.1 I get:
```
>>> np.uint8(255) + np.uint8(255)
<stdin>:1: RuntimeWarning: overflow encountered in scalar add
np.uint8(254)
```
So I think maybe doing this would be a clean way to fix it
```
ne = np.fromfile(fid, UINT8, 3).astype(np.int64)
```
or similar